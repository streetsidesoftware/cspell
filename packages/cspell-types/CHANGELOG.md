# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [5.5.2](https://github.com/streetsidesoftware/cspell/compare/v5.5.1...v5.5.2) (2021-05-30)


### Bug Fixes

* Ignore hash links like `[abcdefa]` ([#1293](https://github.com/streetsidesoftware/cspell/issues/1293)) ([888e25d](https://github.com/streetsidesoftware/cspell/commit/888e25d48254cc57c9fb4f3c5a65d57dfffacff0))





## [5.5.1](https://github.com/streetsidesoftware/cspell/compare/v5.5.0...v5.5.1) (2021-05-29)

**Note:** Version bump only for package @cspell/cspell-types





# [5.5.0](https://github.com/streetsidesoftware/cspell/compare/v5.4.1...v5.5.0) (2021-05-29)


### Features

* Remove incorrect Ignore Hex Digits Regexp ([#1277](https://github.com/streetsidesoftware/cspell/issues/1277)) ([2621eb0](https://github.com/streetsidesoftware/cspell/commit/2621eb02f487d9e466b4936bde8650c338b320b8)), closes [#1276](https://github.com/streetsidesoftware/cspell/issues/1276) [#1276](https://github.com/streetsidesoftware/cspell/issues/1276)





## [5.4.1](https://github.com/streetsidesoftware/cspell/compare/v5.4.0...v5.4.1) (2021-05-11)

**Note:** Version bump only for package @cspell/cspell-types





# [5.4.0](https://github.com/streetsidesoftware/cspell/compare/v5.3.12...v5.4.0) (2021-05-05)

**Note:** Version bump only for package @cspell/cspell-types





## [5.3.12](https://github.com/streetsidesoftware/cspell/compare/v5.3.11...v5.3.12) (2021-04-06)

**Note:** Version bump only for package @cspell/cspell-types





## [5.3.11](https://github.com/streetsidesoftware/cspell/compare/v5.3.10...v5.3.11) (2021-04-03)

**Note:** Version bump only for package @cspell/cspell-types





## [5.3.10](https://github.com/streetsidesoftware/cspell/compare/v5.3.9...v5.3.10) (2021-04-02)

**Note:** Version bump only for package @cspell/cspell-types





## [5.3.8](https://github.com/streetsidesoftware/cspell/compare/v5.3.7...v5.3.8) (2021-03-17)


### Bug Fixes

* update schema generator ([#1074](https://github.com/streetsidesoftware/cspell/issues/1074)) ([528b7d0](https://github.com/streetsidesoftware/cspell/commit/528b7d0b67e1f947832de24e7b66c3d7dfd5fb7f))





## [5.3.7](https://github.com/streetsidesoftware/cspell/compare/v5.3.7-alpha.3...v5.3.7) (2021-03-05)

**Note:** Version bump only for package @cspell/cspell-types





## [5.3.7-alpha.3](https://github.com/streetsidesoftware/cspell/compare/v5.3.7-alpha.2...v5.3.7-alpha.3) (2021-03-05)

**Note:** Version bump only for package @cspell/cspell-types





## [5.3.5](https://github.com/streetsidesoftware/cspell/compare/v5.3.4...v5.3.5) (2021-03-05)


### Bug Fixes

* fix schema pattern for language IDs. ([#1034](https://github.com/streetsidesoftware/cspell/issues/1034)) ([591dde2](https://github.com/streetsidesoftware/cspell/commit/591dde2532a7698f41c5bf824ead226a7ae78cfe))





## [5.3.4](https://github.com/streetsidesoftware/cspell/compare/v5.3.3...v5.3.4) (2021-03-01)

**Note:** Version bump only for package @cspell/cspell-types





## [5.3.3](https://github.com/streetsidesoftware/cspell/compare/v5.3.2...v5.3.3) (2021-02-26)

**Note:** Version bump only for package @cspell/cspell-types





## [5.3.2](https://github.com/streetsidesoftware/cspell/compare/v5.3.1...v5.3.2) (2021-02-26)

**Note:** Version bump only for package @cspell/cspell-types





## [5.3.1](https://github.com/streetsidesoftware/cspell/compare/v5.3.0...v5.3.1) (2021-02-25)

**Note:** Version bump only for package @cspell/cspell-types





# [5.3.0](https://github.com/streetsidesoftware/cspell/compare/v5.3.0-alpha.4...v5.3.0) (2021-02-25)

**Note:** Version bump only for package @cspell/cspell-types





# [5.3.0-alpha.4](https://github.com/streetsidesoftware/cspell/compare/v5.3.0-alpha.3...v5.3.0-alpha.4) (2021-02-25)


### Bug Fixes

* be able to report on glob source. ([#1001](https://github.com/streetsidesoftware/cspell/issues/1001)) ([1020d56](https://github.com/streetsidesoftware/cspell/commit/1020d56ba04e8418d162d4bde8e4948546a8f7d2))





# [5.3.0-alpha.3](https://github.com/streetsidesoftware/cspell/compare/v5.3.0-alpha.2...v5.3.0-alpha.3) (2021-02-23)


### Bug Fixes

* Improve reporting on files matching glob patterns. ([#994](https://github.com/streetsidesoftware/cspell/issues/994)) ([da991f9](https://github.com/streetsidesoftware/cspell/commit/da991f93a061c5b64ce437332c7107ef2ef89472))





# [5.3.0-alpha.2](https://github.com/streetsidesoftware/cspell/compare/v5.3.0-alpha.1...v5.3.0-alpha.2) (2021-02-22)


### Bug Fixes

* Add schema support for `enableFiletypes`. ([#979](https://github.com/streetsidesoftware/cspell/issues/979)) ([d18706e](https://github.com/streetsidesoftware/cspell/commit/d18706e3ffd7a6446396d339b85a753b4ae451e0))
* Add support for custom dictionaries ([#982](https://github.com/streetsidesoftware/cspell/issues/982)) ([196921d](https://github.com/streetsidesoftware/cspell/commit/196921dc635dce7cfdd5b6c300af54beeebde91a))





# [5.3.0-alpha.1](https://github.com/streetsidesoftware/cspell/compare/v5.3.0-alpha.0...v5.3.0-alpha.1) (2021-02-19)


### Bug Fixes

* Fix schema generation to use `deprecatedMessage` ([#972](https://github.com/streetsidesoftware/cspell/issues/972)) ([492dca9](https://github.com/streetsidesoftware/cspell/commit/492dca91466773bdf247fdb87f93d64914d5e3e1))





# [5.3.0-alpha.0](https://github.com/streetsidesoftware/cspell/compare/v5.2.4...v5.3.0-alpha.0) (2021-02-18)


### Features

* Be able to specify files to spell check within the config. ([#948](https://github.com/streetsidesoftware/cspell/issues/948)) ([23f7a48](https://github.com/streetsidesoftware/cspell/commit/23f7a488ef500fb1df5cd234c7d3c2ab4ec02961)), closes [#571](https://github.com/streetsidesoftware/cspell/issues/571)
* Part 1 - glob patterns are relative to the config file. ([#921](https://github.com/streetsidesoftware/cspell/issues/921)) ([a250448](https://github.com/streetsidesoftware/cspell/commit/a2504484ec38f15804cc0a203317266f83566b7c))
* Support local configuration files ([#966](https://github.com/streetsidesoftware/cspell/issues/966)) ([0ccc5fe](https://github.com/streetsidesoftware/cspell/commit/0ccc5fe9eb70ca3a4c6e5a3fc0b653465e76983c))





## [5.2.4](https://github.com/streetsidesoftware/cspell/compare/v5.2.3...v5.2.4) (2021-01-28)

**Note:** Version bump only for package @cspell/cspell-types

<!---
cspell: ignore abcdefa
--->
