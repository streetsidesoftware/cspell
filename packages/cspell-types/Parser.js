module.exports = require('./dist/Parser');
