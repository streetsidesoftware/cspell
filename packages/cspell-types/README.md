# Cspell Types

Contains cspell types and json-schema.

This package contains no dependencies to avoid any security issues.

## Support Future Development

- Become a [<img src="https://github.githubassets.com/images/modules/site/icons/funding_platforms/patreon.svg" width="16" height="16" alt="Patreon">Patreon!](https://patreon.com/streetsidesoftware)

## cspell for enterprise

Available as part of the Tidelift Subscription.

The maintainers of cspell and thousands of other packages are working with Tidelift to deliver commercial support and maintenance for the open source packages you use to build your applications. Save time, reduce risk, and improve code health, while paying the maintainers of the exact packages you use. [Learn more.](https://tidelift.com/subscription/pkg/npm-cspell?utm_source=npm-cspell&utm_medium=referral&utm_campaign=enterprise&utm_term=repo)

## Installation

```
npm i -SD @cspell/cspell-types
```

## Usage

Can be use to make writing `cspell.config.js` files easier.

```js
'use strict';

/** @type { import("@cspell/cspell-types").CSpellUserSettings } */
const cspell = {
  description: 'cspell.config.js file in samples/js-config',
  languageSettings: [
    {
      languageId: 'cpp',
      allowCompoundWords: false,
      patterns: [
        {
          name: 'pound-includes',
          pattern: /^\s*#include.*/g,
        },
      ],
      ignoreRegExpList: ['pound-includes'],
    },
  ],
  dictionaryDefinitions: [
    {
      name: 'custom-words',
      path: './custom-words.txt',
    },
  ],
  dictionaries: ['custom-words'],
};

module.exports = cspell;
```

## API

`CSpellSettings` alias `CSpellUserSettings` is the formal definition of the configuration that controls the spell checker.
