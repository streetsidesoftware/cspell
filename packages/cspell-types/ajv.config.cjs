'use strict';

const keywords = ['deprecationMessage', 'markdownDescription', 'scope'];

function addKeywords(ajv) {
    for (const keyword of keywords) {
        ajv.addKeyword(keyword);
    }
}

module.exports = addKeywords;
