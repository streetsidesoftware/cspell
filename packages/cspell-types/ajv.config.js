'use strict';

const keywords = ['deprecatedMessage', 'markdownDescription', 'scope'];

function addKeywords(ajv) {
    for (const keyword of keywords) {
        ajv.addKeyword(keyword);
    }
}

module.exports = addKeywords;
