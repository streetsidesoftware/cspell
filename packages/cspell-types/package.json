{
  "name": "@cspell/cspell-types",
  "publishConfig": {
    "access": "public"
  },
  "version": "6.6.0",
  "description": "Types for cspell and cspell-lib",
  "type": "commonjs",
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "typesVersions": {
    "*": {
      "*": [
        "dist/*"
      ],
      "dist/index.d.ts": [
        "dist/index.d.ts"
      ],
      "index.d.ts": [
        "dist/index.d.ts"
      ]
    }
  },
  "exports": {
    ".": "./dist/index.js",
    "./Parser": "./dist/Parser.js",
    "./Parser.js": "./dist/Parser.js",
    "./cspell.schema.json": "./cspell.schema.json",
    "./ajv.config": "./ajv.config.js",
    "./ajv.config.js": "./ajv.config.js"
  },
  "scripts": {
    "clean": "rimraf dist coverage .tsbuildinfo",
    "build": "pnpm run compile && pnpm run build-schema",
    "build-dev": "tsc -p tsconfig.dev.json",
    "build-schema": "ts-json-schema-generator --no-top-ref --path src/CSpellSettingsDef.ts --type CSpellSettings --validation-keywords markdownDescription  --validation-keywords scope --validation-keywords deprecated --validation-keywords deprecationMessage -o  ./cspell.schema.json && cp ./cspell.schema.json ../..",
    "clean-build": "pnpm run clean && pnpm run build",
    "compile": "tsc -p .",
    "test-schema": "ajv -s ./cspell.schema.json -d \"cspell.test.{json,yaml}\" -c ./ajv.config.js",
    "test": "jest && pnpm run test-schema",
    "prepare": "echo Skip Building Types"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/streetsidesoftware/cspell.git"
  },
  "files": [
    "dist",
    "ajv.config.js",
    "cspell.schema.json",
    "!**/__mocks__",
    "!**/*.test.*",
    "!**/*.spec.*",
    "!**/*.map"
  ],
  "keywords": [
    "cspell",
    "types"
  ],
  "author": "Jason Dent",
  "license": "MIT",
  "engines": {
    "node": ">=14"
  },
  "devDependencies": {
    "ajv-cli": "^5.0.0",
    "jest": "^28.1.3",
    "rimraf": "^3.0.2",
    "ts-json-schema-generator": "^1.0.0",
    "typescript": "^4.7.4"
  }
}
