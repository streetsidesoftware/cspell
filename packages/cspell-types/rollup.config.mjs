/* eslint-disable node/no-extraneous-import */
import rollupPluginNodeResolve from '@rollup/plugin-node-resolve';
import rollupPluginTypescript from '@rollup/plugin-typescript';
import rollupPluginJson from '@rollup/plugin-json';
import rollupPluginCommonjs from '@rollup/plugin-commonjs';
import { readFileSync } from 'fs';

const pkg = JSON.parse(readFileSync('./package.json', 'utf-8'));

/** @type {import('rollup').RollupOptions} */
const common = {
    input: 'src/index.ts',

    output: {
        sourcemap: true,
    },

    // external: ['@cspell/cspell-pipe', '@cspell/cspell-pipe/sync'],

    treeshake: {
        annotations: true,
        moduleSideEffects: [],
        propertyReadSideEffects: false,
        unknownGlobalSideEffects: false,
    },
};

/**
 * Get new instances of all the common plugins.
 */
function getPlugins(tsconfig = 'tsconfig.esm.json') {
    return [
        rollupPluginTypescript({
            tsconfig,
        }),
        rollupPluginNodeResolve({
            mainFields: ['module', 'exports', 'es', 'es6', 'esm', 'main'],
            extensions: ['.ts', '.js', '.mjs', '.mts', '.node', '.json'],
            preferBuiltins: true,
        }),
        rollupPluginCommonjs({
            transformMixedEsModules: true,
        }),
        rollupPluginJson(),
    ];
}

const plugins = getPlugins();

/** @type {import('rollup').RollupOptions[]} */
const configs = [
    {
        ...common,
        output: [
            { ...common.output, file: pkg.main, format: 'cjs' },
            // { ...common.output, file: pkg.module, format: 'es' },
        ],
        plugins,
    },
    {
        ...common,
        input: 'src/Parser/index.ts',
        output: [
            { ...common.output, file: 'dist/cjs/Parser/index.cjs', format: 'cjs' },
            // { ...common.output, file: pkg.module, format: 'es' },
        ],
        plugins,
    },
];
export default configs;
