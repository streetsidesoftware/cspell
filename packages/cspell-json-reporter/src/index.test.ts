jest.mock('fs', () => ({
    promises: {
        writeFile: async () => undefined,
        mkdir: async () => undefined,
    },
}));

import type { CSpellReporter } from '@cspell/cspell-types';
import { MessageTypes } from '@cspell/cspell-types';
import { promises as fs } from 'fs';
import * as path from 'path';

import { getReporter } from './index';

describe('getReporter', () => {
    let mockWriteFile = jest.spyOn(fs, 'writeFile');

    beforeEach(() => {
        jest.resetAllMocks();
        mockWriteFile = jest.spyOn(fs, 'writeFile');
    });

    afterEach(() => {
        jest.resetAllMocks();
        mockWriteFile.mockReset();
    });

    it('throws for invalid config', () => {
        expect(() => getReporter({ outFile: {} })).toThrowErrorMatchingSnapshot();
    });

    it('saves json to file', async () => {
        const reporter = getReporter({ outFile: 'out.json' });
        await runReporter(reporter);
        expect(mockWriteFile).toHaveBeenCalledTimes(1);

        expect(mockWriteFile.mock.calls[0][0]).toEqual(path.join(process.cwd(), 'out.json'));
        expect(mockWriteFile.mock.calls[0][1]).toMatchSnapshot();
    });

    it.each`
        settings
        ${undefined}
        ${{ outFile: undefined }}
        ${{ outFile: 'stdout' }}
        ${{ outFile: 'stderr' }}
    `('saves json to stdout/stderr $settings', async ({ settings }) => {
        const stdout = jest.spyOn(console, 'log'); // .mockImplementation(() => undefined);
        const stderr = jest.spyOn(console, 'error'); // .mockImplementation(() => undefined);
        const reporter = getReporter(settings);
        await runReporter(reporter);
        expect(joinCalls(stderr.mock.calls)).toMatchSnapshot();
        expect(joinCalls(stdout.mock.calls)).toMatchSnapshot();
    });

    it('saves additional data', async () => {
        const reporter = getReporter({ outFile: 'out.json', verbose: true, debug: true, progress: true });
        await runReporter(reporter);
        expect(fs.writeFile).toHaveBeenCalledTimes(1);
        expect(mockWriteFile.mock.calls[0][0]).toEqual(path.join(process.cwd(), 'out.json'));
        expect(mockWriteFile.mock.calls[0][1]).toMatchSnapshot();
    });
});

// eslint-disable-next-line @typescript-eslint/no-explicit-any
function joinCalls(calls: any[][]): string {
    return calls.map((call) => call.join('<>')).join('\n');
}

async function runReporter(reporter: Required<CSpellReporter>): Promise<void> {
    reporter.debug('foo');
    reporter.debug('bar');

    reporter.error('something went wrong', new Error('oh geez'));

    reporter.info('some logs', MessageTypes.Info);
    reporter.info('some warnings', MessageTypes.Warning);
    reporter.info('some debug logs', MessageTypes.Debug);

    // cSpell:disable
    reporter.issue({
        text: 'fulll',
        offset: 13,
        line: { text: 'This text is fulll of errrorrrs.', offset: 0 },
        row: 1,
        col: 14,
        uri: 'text.txt',
        context: { text: 'This text is fulll of errrorrrs.', offset: 0 },
    });
    // cSpell:enable

    reporter.progress({
        type: 'ProgressFileComplete',
        fileNum: 1,
        fileCount: 1,
        filename: 'text.txt',
        elapsedTimeMs: 349.058747,
        processed: true,
        numErrors: 2,
    });

    await reporter.result({
        files: 1,
        filesWithIssues: new Set(['text.txt']),
        issues: 2,
        errors: 1,
        cachedFiles: 0,
    });
}
