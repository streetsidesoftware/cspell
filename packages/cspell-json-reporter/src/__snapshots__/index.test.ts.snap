// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getReporter saves additional data 1`] = `
"{
    "issues": [
        {
            "text": "fulll",
            "offset": 13,
            "line": {
                "text": "This text is fulll of errrorrrs.",
                "offset": 0
            },
            "row": 1,
            "col": 14,
            "uri": "text.txt",
            "context": {
                "text": "This text is fulll of errrorrrs.",
                "offset": 0
            }
        }
    ],
    "info": [
        {
            "message": "some logs",
            "msgType": "Info"
        },
        {
            "message": "some warnings",
            "msgType": "Warning"
        },
        {
            "message": "some debug logs",
            "msgType": "Debug"
        }
    ],
    "debug": [
        {
            "message": "foo"
        },
        {
            "message": "bar"
        }
    ],
    "error": [
        {
            "message": "something went wrong",
            "error": {}
        }
    ],
    "progress": [
        {
            "type": "ProgressFileComplete",
            "fileNum": 1,
            "fileCount": 1,
            "filename": "text.txt",
            "elapsedTimeMs": 349.058747,
            "processed": true,
            "numErrors": 2
        }
    ],
    "result": {
        "files": 1,
        "filesWithIssues": [
            "text.txt"
        ],
        "issues": 2,
        "errors": 1,
        "cachedFiles": 0
    }
}"
`;

exports[`getReporter saves json to file 1`] = `
"{
    "issues": [
        {
            "text": "fulll",
            "offset": 13,
            "line": {
                "text": "This text is fulll of errrorrrs.",
                "offset": 0
            },
            "row": 1,
            "col": 14,
            "uri": "text.txt",
            "context": {
                "text": "This text is fulll of errrorrrs.",
                "offset": 0
            }
        }
    ],
    "info": [
        {
            "message": "some warnings",
            "msgType": "Warning"
        }
    ],
    "debug": [],
    "error": [
        {
            "message": "something went wrong",
            "error": {}
        }
    ],
    "progress": [],
    "result": {
        "files": 1,
        "filesWithIssues": [
            "text.txt"
        ],
        "issues": 2,
        "errors": 1,
        "cachedFiles": 0
    }
}"
`;

exports[`getReporter saves json to stdout/stderr {"outFile": "stderr"} 1`] = `
"{
    "issues": [
        {
            "text": "fulll",
            "offset": 13,
            "line": {
                "text": "This text is fulll of errrorrrs.",
                "offset": 0
            },
            "row": 1,
            "col": 14,
            "uri": "text.txt",
            "context": {
                "text": "This text is fulll of errrorrrs.",
                "offset": 0
            }
        }
    ],
    "info": [
        {
            "message": "some warnings",
            "msgType": "Warning"
        }
    ],
    "debug": [],
    "error": [
        {
            "message": "something went wrong",
            "error": {}
        }
    ],
    "progress": [],
    "result": {
        "files": 1,
        "filesWithIssues": [
            "text.txt"
        ],
        "issues": 2,
        "errors": 1,
        "cachedFiles": 0
    }
}"
`;

exports[`getReporter saves json to stdout/stderr {"outFile": "stderr"} 2`] = `""`;

exports[`getReporter saves json to stdout/stderr {"outFile": "stdout"} 1`] = `""`;

exports[`getReporter saves json to stdout/stderr {"outFile": "stdout"} 2`] = `
"{
    "issues": [
        {
            "text": "fulll",
            "offset": 13,
            "line": {
                "text": "This text is fulll of errrorrrs.",
                "offset": 0
            },
            "row": 1,
            "col": 14,
            "uri": "text.txt",
            "context": {
                "text": "This text is fulll of errrorrrs.",
                "offset": 0
            }
        }
    ],
    "info": [
        {
            "message": "some warnings",
            "msgType": "Warning"
        }
    ],
    "debug": [],
    "error": [
        {
            "message": "something went wrong",
            "error": {}
        }
    ],
    "progress": [],
    "result": {
        "files": 1,
        "filesWithIssues": [
            "text.txt"
        ],
        "issues": 2,
        "errors": 1,
        "cachedFiles": 0
    }
}"
`;

exports[`getReporter saves json to stdout/stderr {"outFile": undefined} 1`] = `""`;

exports[`getReporter saves json to stdout/stderr {"outFile": undefined} 2`] = `
"{
    "issues": [
        {
            "text": "fulll",
            "offset": 13,
            "line": {
                "text": "This text is fulll of errrorrrs.",
                "offset": 0
            },
            "row": 1,
            "col": 14,
            "uri": "text.txt",
            "context": {
                "text": "This text is fulll of errrorrrs.",
                "offset": 0
            }
        }
    ],
    "info": [
        {
            "message": "some warnings",
            "msgType": "Warning"
        }
    ],
    "debug": [],
    "error": [
        {
            "message": "something went wrong",
            "error": {}
        }
    ],
    "progress": [],
    "result": {
        "files": 1,
        "filesWithIssues": [
            "text.txt"
        ],
        "issues": 2,
        "errors": 1,
        "cachedFiles": 0
    }
}"
`;

exports[`getReporter saves json to stdout/stderr undefined 1`] = `""`;

exports[`getReporter saves json to stdout/stderr undefined 2`] = `
"{
    "issues": [
        {
            "text": "fulll",
            "offset": 13,
            "line": {
                "text": "This text is fulll of errrorrrs.",
                "offset": 0
            },
            "row": 1,
            "col": 14,
            "uri": "text.txt",
            "context": {
                "text": "This text is fulll of errrorrrs.",
                "offset": 0
            }
        }
    ],
    "info": [
        {
            "message": "some warnings",
            "msgType": "Warning"
        }
    ],
    "debug": [],
    "error": [
        {
            "message": "something went wrong",
            "error": {}
        }
    ],
    "progress": [],
    "result": {
        "files": 1,
        "filesWithIssues": [
            "text.txt"
        ],
        "issues": 2,
        "errors": 1,
        "cachedFiles": 0
    }
}"
`;

exports[`getReporter throws for invalid config 1`] = `"cspell-json-reporter settings.outFile must be a string"`;
